packages/powersync_attachments_stream/lib/src/attachment_queue_service.dart

```diff
@@ -0,0 +1,404 @@
+// Implements the attachment queue for PowerSync attachments.
+//
+// This class manages the lifecycle of attachment records, including watching for new attachments,
+// syncing with remote storage, handling uploads, downloads, and deletes, and managing local storage.
+// It provides hooks for error handling, cache management, and custom filename resolution.
+
+import 'dart:async';
+import 'dart:typed_data';
+import 'package:logging/logging.dart';
+import './storage/io_local_storage.dart';
```

Comment:
It would be nice if `AttachmentQueue` were available from `common.dart` and the web, and this import is the only thing preventing that.

It looks like it's only used as a default for `localStorage`. I see it's helpful to not make that a required argument, but I wonder if in this case it may be better to force users to spell out `IOLocalStorage()` when constructing `AttachmentQueue`s to be explicit about the `dart:io` dependency.

I think my preferred API for Dart would be to move the `attachmentsDirectory` parameter into the `IOLocalStorage()` class, I've started a discussion on this offline.